internal/provider/job_resource.go

```diff
@@ -0,0 +1,334 @@
+}
+
+var key_mapping = map[string]string{
+	"inventory":             "inventory",
+	"execution_environment": "execution_environment_id",
+}
+
+func (d *jobResourceModel) GetTemplateId() string {
+	return d.TemplateId.String()
+}
+
+func (d *jobResourceModel) GetURL() string {
+	if !d.URL.IsNull() && !d.URL.IsUnknown() {
+		return d.URL.ValueString()
+	}
+	return ""
+}
+
+func (d *jobResourceModel) ParseHttpResponse(body []byte) error {
+	/* Unmarshal the json string */
+	var result map[string]interface{}
+	err := json.Unmarshal(body, &result)
+	if err != nil {
+		return err
+	}
+
+	d.Type = types.StringValue(result["job_type"].(string))
+	d.URL = types.StringValue(result["url"].(string))
+	d.Status = types.StringValue(result["status"].(string))
+	d.IgnoredFields = types.ListNull(types.StringType)
+
+	if value, ok := result["ignored_fields"]; ok {
+		var keys_list []attr.Value = []attr.Value{}
+		for k := range value.(map[string]interface{}) {
+			key := k
+			if v, ok := key_mapping[k]; ok {
+				key = v
+			}
+			keys_list = append(keys_list, types.StringValue(key))
+		}
+		if len(keys_list) > 0 {
+			d.IgnoredFields, _ = types.ListValue(types.StringType, keys_list)
+		}
+	}
+
+	return nil
+}
+
+func IsValueProvided(value attr.Value) bool {
+	return !value.IsNull() && !value.IsUnknown()
+}
+
+func (d *jobResourceModel) CreateRequestBody() (*bytes.Reader, error) {
+	body := make(map[string]interface{})
+
+	// Extra vars
+	if IsValueProvided(d.ExtraVars) {
+		var extra_vars map[string]interface{}
+		_ = json.Unmarshal([]byte(d.ExtraVars.ValueString()), &extra_vars)
+		body["extra_vars"] = extra_vars
+	}
+
+	// Inventory
+	if IsValueProvided(d.InventoryId) {
+		body["inventory"] = d.InventoryId.ValueInt64()
+	}
+
+	if len(body) == 0 {
+		return nil, nil
+	}
+	json_raw, err := json.Marshal(body)
+	if err != nil {
+		return nil, err
+	}
+	return bytes.NewReader(json_raw), nil
+}
+
+// Configure adds the provider configured client to the data source.
+func (d *JobResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*AAPClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *AAPClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (r JobResource) CreateJob(data JobResourceModelInterface) error {
+
+	// Create new Job from job template
+	req_data, err := data.CreateRequestBody()
+	if err != nil {
+		return err
+	}
+
+	var http_code int
+	var body []byte
+	post_url := "/api/v2/job_templates/" + data.GetTemplateId() + "/launch/"
+	if req_data != nil {
+		http_code, body, err = r.client.doRequest("POST", post_url, req_data)
+	} else {
+		http_code, body, err = r.client.doRequest("POST", post_url, nil)
+	}
```

Comment:
I think this if/else block can be replaced with just 
`http_code, body, err = r.client.doRequest("POST", post_url, req_data)`